modules/siddhi-core/src/main/java/org/wso2/siddhi/core/stream/output/sink/LogSink.java

```diff
@@ -0,0 +1,188 @@
+                                "User has not given a priority so it will be set to default INFO."
+                ),
+                @Example(
+                        syntax = "@sink(type='log'),\n" +
+                                "define stream BarStream (symbol string, price float, volume long)",
+                        description = "In this example BarStream uses log sink. The user has not given prefix or " +
+                                "priority so they will be set to their default values."
+                )
+        }
+)
+
+public class LogSink extends Sink {
+    private static final Logger logger = Logger.getLogger(LogSink.class);
+    private static final String PREFIX = "prefix";
+    private static final String PRIORITY = "priority";
+    private String logPrefix;
+    private LogPriority logPriority;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{Object.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    @Override
+    protected void init(StreamDefinition outputStreamDefinition, OptionHolder optionHolder,
+                        ConfigReader sinkConfigReader, SiddhiAppContext siddhiAppContext) {
+        String defaultPrefix = siddhiAppContext.getName() + " : " + outputStreamDefinition.getId();
+        logPrefix = optionHolder.validateAndGetStaticValue(PREFIX, defaultPrefix);
+        logPriority = LogPriority.valueOf(optionHolder.validateAndGetStaticValue(PRIORITY, "INFO")
+                .toUpperCase());
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions transportOptions) throws ConnectionUnavailableException {
+        String message;
+        if (payload instanceof Object[]) {
+            message = logPrefix + " : " + Arrays.deepToString((Object[]) payload);
+        } else {
+            message = logPrefix + " : " + payload;
+        }
+        logMessage(logPriority, message);
+    }
+
+    private void logMessage(LogSink.LogPriority logPriority, String message) {
+        switch (logPriority) {
+            case INFO:
+                logger.info(message);
+                break;
+            case DEBUG:
+                logger.debug(message);
+                break;
+            case WARN:
+                logger.warn(message);
+                break;
+            case FATAL:
+                logger.fatal(message);
+                break;
+            case ERROR:
+                logger.error(message);
+                break;
+            case OFF:
+                break;
+            case TRACE:
+                logger.trace(message);
+                break;
+        }
+    }
+
+    enum LogPriority {
+        INFO,
+        DEBUG,
+        WARN,
+        FATAL,
+        ERROR,
+        OFF,
+        TRACE
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+
```

Comment:
Why empty methods?

Reply:
because this implements sink abstract class and these methods need to be overridden

Reply:
if these methods are not supposed to be invoked, then throw an `UnsupportedOperationException`. Otherwise remove the empty line.

Reply:
they will be invoked but this particular sink doesnt have to do anything. i will remove the empty line